### PR TITLE
Generate activation "token"

### PR DESCRIPTION
**Pull request by ppiotr191**
Generate activation "token" for user activation. For for random key, i used uuid.

### CODE REVIEW

File: src/main/java/com/intive/patronage/toz/users/UserActivationService.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Date;
+import java.util.UUID;
+
+@Service
+public class UserActivationService {
+
+    private final static String USER_ACTIVATION = "User Activation";
+    private final static String USER = "User";
```
Comment (macieg-b) [inline src/main/java/com/intive/patronage/toz/users/UserActivationService.java]: USER is never used, is it necessary?

File: src/main/java/com/intive/patronage/toz/users/UserActivationService.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    public UserActivation createUserActivation(){
+        UserActivation userActivation = new UserActivation();
+        Date date = new Date();
+        userActivation.setExpiredDate(new Date(date.getTime() + expiration));
+        userActivation.setIsActivated(false);
+        userActivationRepository.save(userActivation);
+        return userActivation;
+    }
```
Comment (mlotysz) [inline src/main/java/com/intive/patronage/toz/users/UserActivationService.java]: Add EOL

File: src/test/java/com/intive/patronage/toz/users/UserActivationServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    @DataProvider
+    public static Object[] getProperExpiredUserActivation() {
+        UserActivation expiredUserActivation = new UserActivation();
+        expiredUserActivation.setId(EXPIRED_EXPECTED_ID);
+        expiredUserActivation.setCreated(CURRENT_DATE);
+        expiredUserActivation.setExpiredDate(EXPIRED_EXPECTED_CREATED_DATE);
+        expiredUserActivation.setIsActivated(false);
+        return new UserActivation[]{expiredUserActivation};
+    }
+
```
Comment (mlotysz) [inline src/test/java/com/intive/patronage/toz/users/UserActivationServiceTest.java]: Remove EOL. Use one to separate methods. (Fix other places too.)

**Comment by mlotysz** [comment]
Maybe you forgot, that we planned to use JWT token to handle activations too. You generate JWT token with exp time and pass it to user. Only what need to do it to allow user creation for reruest which contains valid JWT token. It's needs to wait until JWT token will be merged into master...

File: src/main/java/com/intive/patronage/toz/users/UserActivationService.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    @Autowired
+    public UserActivationService(UserActivationRepository userActivationRepository, UserRepository userRepository){
+        this.userRepository = userRepository;
+        this.userActivationRepository = userActivationRepository;
+    }
+
+    public UserActivation createUserActivation(){
+        UserActivation userActivation = new UserActivation();
+        Date date = new Date();
+        userActivation.setExpiredDate(new Date(date.getTime() + expiration));
```
Comment (LukaszPiskadlo) [inline src/main/java/com/intive/patronage/toz/users/UserActivationService.java]: Isn't expiration date set to low? Token will expire after less than 3 minutes. Date.getTime() returns time in milliseconds.

**Pull request closed without merging**